buildSrc/src/main/java/gradlebuild/NcursesRuntimePlugin.java

```diff
@@ -0,0 +1,109 @@
+package gradlebuild;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import dev.nokee.platform.base.BuildVariant;
+import dev.nokee.platform.base.Variant;
+import dev.nokee.platform.jni.JavaNativeInterfaceLibrary;
+import dev.nokee.runtime.nativebase.TargetMachineFactory;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.plugins.ExtensionAware;
+import org.gradle.api.provider.Provider;
+import org.gradle.api.provider.SetProperty;
+import org.gradle.api.reflect.TypeOf;
+import org.gradle.api.specs.Spec;
+import org.gradle.nativeplatform.platform.OperatingSystem;
+import org.gradle.nativeplatform.platform.internal.DefaultNativePlatform;
+
+import java.io.File;
+import java.util.Set;
+import java.util.concurrent.Callable;
+
+import static gradlebuild.JavaNativeInterfaceLibraryUtils.library;
+import static gradlebuild.NcursesVersion.NCURSES_5;
+import static gradlebuild.NcursesVersion.NCURSES_6;
+import static java.util.Arrays.stream;
+
+@SuppressWarnings("UnstableApiUsage")
+public class NcursesRuntimePlugin implements Plugin<Project> {
+    @Override
+    public void apply(Project project) {
+        project.getExtensions().add(new TypeOf<Provider<Set<NcursesVersion>>>() {}, "availableNcursesVersions", availableNcursesVersions(project));
+
+        project.getPluginManager().withPlugin("dev.nokee.jni-library", appliedPlugin -> {
+            library(project, library -> {
+                // Register ncurses dimension
+                SetProperty<NcursesVersion> ncurses = registerNcursesDimension(library);
+
+                // Defaults to all ncurses versions
+                ncurses.convention(ImmutableSet.copyOf(NcursesVersion.values()));
+
+                // Register dimension as library extensions
+                ((ExtensionAware) library).getExtensions().add("targetNcurses", ncurses);
```

Comment:
❓ I'm still waiting for guidance on the usage for `ExtensionAware` by plugin authors. See https://gradle.slack.com/archives/C02FZQ5EZDE/p1638225105038300